File: Python/BalancedBinaryTree/BalancedBinaryTree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
	# (112ms)
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        def height(root):
	        if not root:
			    return 0
			return 1 + max(height(root.left), height(root.right))
		if not root:
		    return True
		return abs(height(root.left) - height(root.right)) <= 1 
		and self.isBalanced(root.left) and self.isBalanced(root.right)

File: JavaScript/BalancedBinaryTree/BalancedBinaryTree.js
/**
 * Created by Phoenix-Z on 2017/4/9.
 */
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * (172ms)
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    if(root === null)
        return true;
    return Math.abs(height(root.left) - height(root.right)) <= 1
        && isBalanced(root.left) && isBalanced(root.right);
};

var height = function (root) {
    if(root === null)
        return 0;
    return 1 + Math.max(height(root.left), height(root.right));
};